### DISCUSSION TITLE
Unique keycloak sessions with multiple tabs?

### DISCUSSION THREAD

**Discussion by rkozura** [keycloak/keycloak#28309] [Q&A]
Hello,

I am using `keycloak-js` and am wondering if this is possible.

I am trying to sign in using a dynamic scope, something like `key:foo`.

After sign in and receiving a token, another tab is opened on the same browser.  Since I have cookies enabled, I am automatically logged into my application.  This second authentication uses a different scope: `key:bar`

But only one session is created between tabs.  Whenever a refresh token is retrieved, I am always getting back a token for the `key:bar` scope (the latest login attempt)

This means that the first tab `key:foo` now has a token that is only good for `key:bar`.

Is there a way to disable this behavior?

**Comment by codespearhead** [comment]
It's possible if you create multiple KeycloakJS clients and customize the key of the token when saving it in [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).

Conversely, you can use [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage), which has the session isolated on a page (tab) basis, with the caveat that it's cleared when the page is closed.

**Reply by robkozura** [reply]
@codespearhead I am using sessionStorage, however the sessions match between the tabs.  The second tab that I open is basically sharing the same session as the one in Keycloak.  I believe this is due to the following cookies that enable SSO:

`AUTH_SESSION_ID`, `KEYCLOAK_SESSION`, `KEYCLOAK_SESSION_LEGACY` etc.

If there was a way to disable SSO completely, I think my issue would be resolved.

**Reply by codespearhead** [reply]
@robkozura  I agree.

Try doing this:  `${realm-name} > Authentication > Flow name: browser > Steps: Cookie` and set it to `disabled`.

it worked on my end, but confirmation is never enough.

Besides, if you confirm that works, then issue #13789 can be closed.

**Reply by robkozura** [reply]
@codespearhead 

I do get challenged for authentication again when I open a new tab, however there is still only 1 session for my user.  I am using IDP FWIW.

I was expecting one session for each tab with the Cookie step disabled.
